### PR TITLE
Update port library's processor detection APIs and enums

### PR DESCRIPTION
**Pull request by harryyu1994**
This change involves updating the omrsysinfo_get_x86_description() API and
a few enums.

Issue: #4339

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
The purpose of this change is to move x86's processor feature detection out of codegen and into the port library. I referenced this file(https://github.com/eclipse/omr/blob/master/compiler/x/runtime/X86Runtime.hpp) for the change.

**Comment by harryyu1994** [comment]
@0xdaryl @andrewcraik I've seen you guys review related changes before, could you guys take a look?

**Comment by harryyu1994** [comment]
@rwy0717 @0xdaryl @youngar @mstoodle @charliegracie Could anyone help review this one? thanks!

File: port/common/omrsysinfo_helpers.c [hunk 1]
```diff
@@ -107,20 +140,41 @@ omrsysinfo_get_x86_description(struct OMRPortLibrary *portLibrary, OMRProcessorD
...
-				desc->processor = OMR_PROCESSOR_X86_INTELNEHALEM;
-			} else if (totalModelCode == CPUID_MODELCODE_INTELCORE2) {
-				desc->processor = OMR_PROCESSOR_X86_INTELCORE2;
-			} else {
-				desc->processor = OMR_PROCESSOR_X86_INTELP6;
+			uint32_t totalModelCode = modelCode + (extendedModelCode << 4);
+
+			switch (totalModelCode) {
+			case CPUID_MODELCODE_INTELSKYLAKE:
+			    desc->processor = OMR_PROCESSOR_X86_INTELSKYLAKE;
```
[the remark this answers is not in the document]
Comment (harryyu1994) [reply port/common/omrsysinfo_helpers.c]: Fixed. Just wondering, how did you tell that I used a mix of tabs and spaces?

[the remark this answers is not in the document]
Comment (rwy0717) [reply port/common/omrsysinfo_helpers.c]: I use a chrome extension called [refined github](https://github.com/sindresorhus/refined-github), which allows me to see whitespace. It's a pretty useful tool, but not everything it does is good. I've had to disable quite a few "features".

File: port/common/omrsysinfo_helpers.c [hunk 2]
```diff
@@ -32,43 +32,76 @@
...
-#define CPUID_MODELCODE_INTELCORE2          0x0F
-
-#define CPUID_FAMILYCODE_AMDKSERIES         0x05
-#define CPUID_FAMILYCODE_AMDATHLON          0x06
-#define CPUID_FAMILYCODE_AMDOPTERON         0x0F
-
-#define CPUID_MODELCODE_AMDK5               0x04
-
-static void omrsysinfo_get_x86_cpuid(uint32_t leaf, uint32_t *cpuInfo);
+#define CPUID_EXA                                         0
```
Comment (rwy0717) [inline port/common/omrsysinfo_helpers.c]: Should `EXA` be called `EAX`?

Comment (harryyu1994) [reply port/common/omrsysinfo_helpers.c]: Yes, that was a typo.

**Comment by harryyu1994** [comment]
I introduced another small change in "port/unix/omrsysinfo.c" that fixes a few Macro typos.

**Comment by harryyu1994** [comment]
Attaching the list of references used for this change, I figured this may help the reviewers:
- https://github.com/eclipse/omr/blob/master/compiler/x/runtime/X86Runtime.hpp#L27-L110
- https://github.com/eclipse/omr/blob/master/compiler/x/codegen/OMRCodeGenerator.cpp#L112-L201
- https://github.com/eclipse/omr/blob/master/compiler/env/Processors.hpp#L63-L164
- https://github.com/eclipse/omr/blob/master/compiler/env/ProcessorInfo.hpp#L189-L212

OpenJ9's Processor Detection on x86 is all-over the place, there are 3 different variations in the code base with only one being up-to-date. OpenJ9 codegen's implementation seems to be the up-to-date and is what I referenced(mostly).
This change unifies everything into one location.

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by harryyu1994** [comment]
```bash
13:58:48  /home/jenkins/workspace/PullRequest-linux_390-64/port/common/omrsysinfo_helpers.c:41:19: fatal error: cpuid.h: No such file or directory
13:58:48   #include <cpuid.h>
13:58:48                     ^
13:58:48  compilation terminated.
```

Looking into it..

**Comment by harryyu1994** [comment]
@rwy0717 Could we re-launch the builds/tests?

**Comment by rwy0717** [comment]
@genie-omr build all

**Comment by harryyu1994** [comment]
@rwy0717 Thanks for the review, could this be merged now? Also, I'm wondering if you could review this short PR too(https://github.com/eclipse/omr/pull/4838). That one is related to port library cpu detection as well.

**Pull request merged**